extension/src/repository/data/index.ts

```diff
@@ -20,13 +19,12 @@ import {
   EXPERIMENTS_GIT_REFS
 } from '../../experiments/data/constants'
 import { DeferredDisposable } from '../../class/deferred'
+import { Flag } from '../../cli/constants'
 
 export type Data = {
-  diffFromHead: DiffOutput
-  diffFromCache: StatusOutput
-  untracked: Set<string>
+  dataStatus: DataStatusOutput
   hasGitChanges: boolean
-  tracked?: ListOutput[]
+  untracked: Set<string>
```

Comment:
[F] If we get https://github.com/iterative/dvc/issues/8052 fixed up then we will be able to avoid this call to git (we'll be able to pull out even more code). I've set things up so that we can move in that direction easily.